notebooks/working/3.0-mab-independent-variable-simulation.md

```diff
@@ -34,487 +47,269 @@ import random
-```
+    # simulate the availability matrix based on number 
+    # of available alternatives
+    N = len(alt_name_dict)
+    av_sim = [np.array([1] * K + [0]*(N-K)) for K in alts_sim]
 
-```python
-# # Define the nature of each variables whether
-# # discrete/categorical or continuous
-# variable_type = {'num_kids': 'categorical',
-#                  'household_size': 'categorical',
-#                  'num_cars': 'discrete',
-#                  'num_licensed_drivers': 'categorical',
-#                  'cross_bay': 'categorical',
-#                  'total_travel_time_drive_alone': 'continuous',
-#                  'total_travel_time_shared_2': 'continuous',
-#                  'total_travel_time_shared_3p': 'continuous',
-#                  'total_travel_time_wtw': 'continuous',
-#                  'total_travel_time_dtw': 'continuous',
-#                  'total_travel_time_wtd': 'continuous',
-#                  'total_travel_time_walk': 'continuous',
-#                  'total_travel_time_bike': 'continuous',
-#                  'total_travel_distance_drive_alone': 'continuous',
-#                  'total_travel_distance_shared_2': 'continuous',
-#                  'total_travel_distance_shared_3p': 'continuous',
-#                  'total_travel_distance_wtw': 'continuous',
-#                  'total_travel_distance_dtw': 'continuous',
-#                  'total_travel_distance_wtd': 'continuous',
-#                  'total_travel_distance_walk': 'continuous',
-#                  'total_travel_distance_bike': 'continuous',
-#                  'total_travel_cost_drive_alone': 'continuous',
-#                  'total_travel_cost_shared_2': 'continuous',
-#                  'total_travel_cost_shared_3p': 'continuous'}
-```
+    # Shuffle the available alternatives for each observation
+    # because av_sim will always start with 1s
+    for x in av_sim:
+        np.random.shuffle(x)
 
-```python
-# # Simulate dataframe based on the estimated distributions
-# sim_bike_data = bike_data_fitter.SimDf(size=5000)
+    # Shuffle the availability across different observations
+    np.random.shuffle(av_sim)
+    
+    # Create columns for the availability matrix
+    AV_columns = [alt_name_dict[i]+'_AV' for i in alt_name_dict.keys()]
+    
+    # Create alternative availability matrix with AV_columns
+    AV_Df = pd.DataFrame(data=av_sim, columns=AV_columns)
+    return AV_Df
+
+# Function to generate fake choice column
+# this functionality will be relocated to
+# a different function, most likely a function
+# that merges availability matrix and simulated
+# data or a function that converts to long_data
+def createFakeChoiceCol(AV_matrix):
+    # Create an random choice column based on available
+    # alternatives for each observation - This column will
+    # be needed when converting to long data -- this will
+    # be moved to a different column
+    fake_choice = [random.choice(np.nonzero(a == 1)[0]) + 1 for a in np.array(AV_matrix)]
```

Comment:
I find this line hard to understand upon first reading it.

I would separate the various parts of this line to show what one desires with each action. For example,
```suggestion
    available_alt_indices = lambda a: np.nonzero(a == 1)[0]
    fake_choice = [random.choice(available_alt_indices(a)) + 1 for a in np.array(AV_matrix)]
```

Reply:
Very true.